### PR TITLE
fix: fix db representation so that users can be accessed and removed properly

### PR DESCRIPTION
**Pull request by Lauraducky**

### CODE REVIEW

File: src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs [hunk 1]
```diff
@@ -18,24 +18,23 @@ public PockyUserRepository(DatabaseContext context)
...
         {
             return _context.PockyUsers
-                .Where(x => x.UserId == userId)
                 .Include(x => x.PegsGiven)
                     .ThenInclude(x => x.Receiver)
                 .Include(x => x.PegsReceived)
                 .Include(x => x.Location)
                 .Include(x => x.Roles)
-                .SingleOrDefault();
+                .SingleOrDefault(x => x.UserId == userId);
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs]: what does this change do?

File: src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs [hunk 2]
```diff
@@ -18,24 +18,23 @@ public PockyUserRepository(DatabaseContext context)
...
         {
             return _context.PockyUsers
-                .Where(x => x.Username == username)
                 .Include(x => x.PegsGiven)
-                .ThenInclude(x => x.Receiver)
+                    .ThenInclude(x => x.Receiver)
                 .Include(x => x.PegsReceived)
                 .Include(x => x.Location)
                 .Include(x => x.Roles)
+                .Where(x => x.Username == username)
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs]: and this one?

File: src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs (hunk 1 above)
Comment (Evangelinexx) [reply src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs]: It moves the logic from the where into the singleordefault.

I changed it to use a where for performance reasons, so I think we should move it back. The reason is you want to do the filtering before you do any table joins, as table joins are expensive

File: src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs (hunk 2 above)
Comment (Evangelinexx) [reply src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs]: Agreed on this one too, we should modify the PR to just include the databasecontext change

File: src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs (hunk 1 above)
Comment (Lauraducky) [reply src/PockyBot.NET/Persistence/Repositories/PockyUserRepository.cs]: Honestly I made these changes when I was trying to diagnose the issue, no problems with changing them back

**Pull request merged**